src/js/common/schemaform/widgets/ArrayCountWidget.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+// We're expanding or contracting the array based on the count
+// and returning undefined if the array should be empty
+function getValue(count, value = []) {
+  if (!count || parseInt(count, 10) < 1) {
+    return undefined;
+  }
+
+  const intCount = parseInt(count, 10);
+  if (intCount < value.length) {
+    return value.slice(0, intCount);
+  }
+
+  return value.concat(Array(intCount - value.length).fill({}));
+}
+
+export default class ArrayCountWidget extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: !!props.value ? props.value.length : undefined };
+  }
+
+  // Need to keep the count state in sync, but don't want to overwrite if the
+  // user has chosen '' or 0
+  componentWillReceiveProps(props) {
+    const arrayData = props.value || [];
+    if (arrayData.length && this.state.value !== arrayData.length) {
+      this.setState({ value: arrayData.length });
+    } else if (!this.state.value && !!arrayData.length) {
+      this.setState({ value: arrayData.length });
+    }
+  }
+
+  updateArrayLength = (event) => {
+    const count = event.target.value;
+    this.setState({ value: count }, () => {
```

Comment:
That's a good point. I gave it another shot with `componentDidUpdate`, but it leads to calling `onChange` twice on every change and it's hard to wrap your head around the logic to try to avoid that.